krita_batch_exporter/Infrastructure.py

```diff
@@ -14,8 +14,46 @@
 
 
 def dataToPIL(wnode):
-    img = wnode.node.projectionPixelData(*wnode.bounds).data()
-    img = Image.frombytes("RGBA", wnode.size, img, "raw", "BGRA", 0, 1)
+    """
+    Returns an Image object using the Pillow library with 8-bit sRGB data that Pillow can then process.
+    If the input node's color space isn't 8-bit sRGB, uses Krita to convert the document to the right color space.
+    """
+
+    SRGB_PROFILE = "sRGB-elle-V2-srgbtrc.icc"
```

Comment:
If we have that info, there's a much more robust solution then:

```is_srgb = "srgb" in wnode.node.colorProfile().lower()```

Reply:
I'm not sure about that. That's just a filename. Not all profiles with "srgb" as a substring of their filenames (or even worse, paths) are certain to be actual sRGB profiles. It's better to err on the side of caution, imo, especially since Elle's profile is the default, and will be the one most Krita users are using, anyway. There is really no need to risk the false positives.

Reply:
To have a false positive, you'd need a non-srgb color profile to include srgb in its name, which isn't technically impossible but would be real strange.
I checked in Krita, and all srgb profiles have srgb in their name. so the code above should work.

Checking for an exact string is the best way to have to do maintenance later: it just takes for one character to change, even just capitalization or symbol, for the code to not work anymore.

Reply:
> all srgb profiles have srgb in their name. so the code above should work.

Except that not only sRGB profiles have `srgb` in their name. I checked in Krita and there is `ClayRGB-elle-V4-srgbtrc.icc` which is not a sRGB profile, but has `srgb` in its name. In Elle's naming convention, `srgbtrc` indicated that a profile had a sRGB power curve: it does not mean that it necessarily had a sRGB color gamut.

You see, for false positives to not occur, you need to prove "all profiles with `srgb` in their names are sRGB", not "all sRGB profiles have `srgb` in their name". That's just basic logic.

> Checking for an exact string is the best way to have to do maintenance later: it just takes for one character to change

I'd say that outputting bad data is the best way to have a non-working plugin.

> even just capitalization or symbol

Well, I have made the comparison case-insensitive, but please note that in case the default profile's name is changed, the code would stop working anyway, because that's also the profile we're converting to.

Reply:
> You see, for false positives to not occur, you need to prove "all profiles with srgb in their names are sRGB"

You're right about the logic part! I checked the source code, all profiles that have the string `sRGB` (with that caps) in their name are srgb. Either they start with that, or have it in their first part of the file name.

But I also didn't think about the gamma curves of the profiles, that are different.

Thanks!